.github/workflows/soak-testing.yml

```diff
@@ -0,0 +1,233 @@
+# - LOGS_NAMESPACE? + LOGS_GROUP
+
+name: Soak Testing
+on:
+  schedule:
+    - cron: '0 */6 * * *' # Every 6 hours
+  # push:
+  #   branches: [ main ]
+env:
+  # Language Specific
+  PROCESS_COMMAND_LINE: /usr/local/bin/python3 application.py
+  PROCESS_EXECUTABLE_NAME: python3
+  # - Every language must also set the path for the sample-app in the `Build and Push Docker image` step.
+  # Global Env Vars
+  AWS_REGION: us-west-2
+  SOAK_TEST_DURATION_IN_MINUTES: 300
+  HOSTMETRICS_COLLECTION_INTERVAL_IN_SECONDS: 600
+  CPU_LOAD_THRESHOLD: 75
+  TOTAL_MEMORY_THRESHOLD: 1610612736
+  # Add this?
+  # RATE: TPS
+
+jobs:
+  test_apps_and_publish_results:
+    name: Publish app and Soak Test - (${{ matrix.app-platform }}, ${{ matrix.instrumentation-type }})
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      packages: write
+      issues: write
+    strategy:
+      fail-fast: false
+      matrix:
+        app-platform: [ flask ]
+        # instrumentation-type: [ auto, manual ]
+        instrumentation-type: [ auto ]
+    env:
+      IMAGE_SUFFIX: sample-app-${{ matrix.app-platform }}-${{ matrix.instrumentation-type }}-${{ github.sha }}
+    steps:
+      - name: Configure Soak Test environment variables
+        # TODO: Does the LOGS_NAMESPACE need a `-$GITHUB_RUN_ID`?
+        run: >-
+          echo "LOGS_NAMESPACE=${{ github.repository }}/soak-tests" >> $GITHUB_ENV;
+          echo "NUM_OF_CPUS=$(nproc --all)" >> $GITHUB_ENV;
+      - uses: actions/checkout@v2
+      - name: Log in to the GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+      - name: Extract metadata (tags, labels) for Docker
+        id: docker-image-metadata
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ghcr.io/${{ github.repository }}-${{ env.IMAGE_SUFFIX }}
+      # TODO: Could we just build in `docker-compose` build/context without publishing?
+      - name: Build and Push Docker image
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          context: integration-test-apps/${{ matrix.instrumentation-type}}-instrumentation/${{ matrix.app-platform }}
+          tags: ${{ steps.docker-image-metadata.outputs.tags }}
+          labels: ${{ steps.docker-image-metadata.outputs.labels }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
```

Comment:
Use same pattern as

https://github.com/aws-observability/aws-otel-python/blob/main/.github/workflows/integration-testing.yml#L28

Reply:
Do you mean we should use the `us-east-1` region?

Otherwise, if you mean the `role-to-assume` key, the role was giving me a headache for permissions and I found a simple policy could work. But I'll look deeper into the way our repos do it now that I know the end result of what I want.

Reply:
> Yeah we use the separate role to authenticate our CI's right now.

I'll look into adding a role then for consistency. Although I would say the permissions are pretty strict and pretty harmless that a role doesn't make sense? From what I read on the docs usually roles are given to AWS services to access other AWS services, not as much for GitHub like we need here.

> To confirm, right now these workflows won't work in this repo because the creds haven't been set up right? We'll just add to the terraform

Correct, that's still a TODO for this PR.

Reply:
Okay I read the [actions docs](https://github.com/aws-actions/configure-aws-credentials#assuming-a-role) and it should be simple to follow the same pattern as you said.

We will just add my permissions to the necessary role in the comment you have above, so I updated this PR with the only TODO to update the terraform.

One call-out is that I set the role to last for 6 hours because that's how long the Soak Test can last.

Reply:
I noticed that the `assume role max duration` variable [is set to be 21600 in this file](https://github.com/aws-observability/aws-otel-ops/blob/ac535258130f2e3110d06bf3e467aa7a13375d14/lambda.tf#L14) so since the files on the top level get combined I assume it should work for my case too!

Reply:
Looks like [the workflow failed](https://github.com/aws-observability/aws-otel-python/runs/3637638145?check_suite_focus=true) because the max duration does not allow for the 6 hours (21600 seconds) needed here. We'll need to update those permissions.